compiler/luci/pass/src/QuantizeWithMinMaxPass.cpp

```diff
@@ -729,6 +729,14 @@ struct QuantizeActivation final : public luci::CircleNodeMutableVisitor<bool>
         circle_node->quantparam()->scale.push_back(scaling_factor);
         circle_node->quantparam()->zerop.push_back(zp);
       }
+      // Fix special attributes
+      if (circle_node->opcode() == luci::CircleOpcode::CAST)
+      {
+        auto *cast = loco::must_cast<luci::CircleCast *>(circle_node);
+        auto *cast_input = loco::must_cast<luci::CircleNode *>(cast->x());
+        cast->in_data_type(cast_input->dtype());
```

Comment:
For cast (fp32->fp32), if `cast_input` is quantized later than `cast`, `cast_input->dtype()` may be fp32, which is invalid.

Cast (fp32->fp32) seems unnatural, so it would be good to add an assert `assert(cast_input->dtype() != float32)`.

Reply:
> if cast_input is quantized later than cast

Currently this code is called in loop over topologically sorted nodes, so described situation should be impossible. (I assume that sorting algorithm is correct).

If this visitor should be agnostic to the order, I'll move this code to separate loop and leave a comment that it should be always sorted.

P.s. I'll add assert for fp32->fp32. It is indeed suspicious case.